### PR TITLE
Update IBM Watson ML sample to use V4 api

### PR DESCRIPTION
**Pull request by kevinyu98**
This PR contains the following updates:

- update `train/store/deploy ` under ibm-component watson to use WML v4 api
- update   ibm-samples' `watson_train_serve_pipeline`

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/3274)

### CODE REVIEW

**Comment by kevinyu98** [comment]
@animeshsingh @Tomcli

File: components/ibm-components/watson/train/src/wml-train.py [hunk 1]
```diff
@@ -68,73 +71,117 @@ def train(args):
                        "instance_id": wml_instance_id,
                        "apikey": wml_apikey
                       }
-    client = WatsonMachineLearningAPIClient( wml_credentials )
-
+    client = WatsonMachineLearningAPIClient ( wml_credentials )
```
Comment (Tomcli) [inline components/ibm-components/watson/train/src/wml-train.py]:
```suggestion
    client = WatsonMachineLearningAPIClient(wml_credentials)
```

File: components/ibm-components/watson/train/src/wml-train.py [hunk 2]
```diff
@@ -68,73 +71,117 @@ def train(args):
...
+                "inputs": [],
+                "outputs": [],
+                "parameters": {
+                    "name": "tf-mnist",
+                    "description": wml_run_definition,
+                    "command": wml_execution_command,
+                    "training_lib_href": "/v4/libraries/"+custom_library_uid,
+                    "compute": {
+                        "name": "k80",
+                        "nodes": 1
```
Comment (Tomcli) [inline components/ibm-components/watson/train/src/wml-train.py]: Maybe add new parameters for the name and node so users can change their t-shirt size?

File: components/ibm-components/watson/train/src/wml-train.py (hunk 1 above)
Comment (kevinyu98) [reply components/ibm-components/watson/train/src/wml-train.py]: changed.

File: components/ibm-components/watson/train/component.yaml [hunk 3]
```diff
@@ -26,6 +26,8 @@ inputs:
   - {name: run_definition,    description: 'Name for the Watson Machine Learning training definition', default: 'python-tensorflow-definition'}
   - {name: run_name,          description: 'Name for the Watson Machine Learning training-runs', default: 'python-tensorflow-run'}
   - {name: author_name,       description: 'Name of this training job author', default: 'default-author'}
+  - {name: compute_name,      description: 'Name of the compute tiers, in WML is the gpu count', default: 'k8'}
```
Comment (Tomcli) [inline components/ibm-components/watson/train/component.yaml]: default should be k80

File: components/ibm-components/watson/train/src/wml-train.py [hunk 4]
```diff
@@ -103,8 +106,8 @@ def train(args):
                     "command": wml_execution_command,
                     "training_lib_href": "/v4/libraries/"+custom_library_uid,
                     "compute": {
-                        "name": "k80",
-                        "nodes": 1
+                        "name": wml_runtime_version_v4,
```
Comment (Tomcli) [inline components/ibm-components/watson/train/src/wml-train.py]: it should be `wml_compute_name`

File: components/ibm-components/watson/train/src/wml-train.py (hunk 2 above)
Comment (kevinyu98) [reply components/ibm-components/watson/train/src/wml-train.py]: done

File: components/ibm-components/watson/train/component.yaml (hunk 3 above)
Comment (kevinyu98) [reply components/ibm-components/watson/train/component.yaml]: thanks, done

File: components/ibm-components/watson/train/src/wml-train.py (hunk 4 above)
Comment (kevinyu98) [reply components/ibm-components/watson/train/src/wml-train.py]: done, pushed new image also.

**Pull request merged**